### PR TITLE
Create an Index and Store Documents

### PR DESCRIPTION
**Pull request by zulander1**

### CODE REVIEW

**Comment by rtynski** [comment]
Suggestion:
- Delete a separate directory for the project (as other)
- All models to Models directory and in separate files.
- Create helpers for init data

**Comment by zulander1** [comment]
@rtynski what do you mean by Delete a separate directory for the project (as other)? Also, I've applied your suggestions

**Comment by rtynski** [comment]
@zulander1 
FROM:
examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore.sln
examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore.csproj
examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/*.cs
TO:
examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore.sln (this file is I think optional)
examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore.csproj
examples/Redis.OM.CreateIndexStore/*.cs

Look: examples/Redis.OM.BasicMatchingQueries/

**Comment by SuzeShardlow** [comment]
Thanks @zulander1, the reviewer for this will be @slorello89.

**Comment by SuzeShardlow** [comment]
Closes #218 when merged.

File: examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Program.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿using Redis.OM;
+using Redis.OM.CreateIndexStore;
+using Redis.OM.CreateIndexStore.Models;
+
+#region Setup provides,indexes and seed
+
+var provider = new RedisConnectionProvider("redis://172.29.252.187:6379");
```
Comment (slorello89) [inline examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Program.cs:7]: move this to `redis://localhost:6379` - for the moment I think the only assumption we'll make about connectivity is that they have a local instance of redis running.

File: examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Program.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+await SeedDataHelpers.SeedEmployess(employee);
+await SeedDataHelpers.SeedCustomers(customers);
+await SeedDataHelpers.SeedStore(stores);
+
+#endregion Setup provides,indexes and seed
+
+#region Basic queries
+
+IList<Store> allStores = await stores.ToListAsync();
```
Comment (slorello89) [inline examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Program.cs:25]: Move all these declarations from explicit -> var

File: examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Program.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+﻿using Redis.OM;
+using Redis.OM.CreateIndexStore;
+using Redis.OM.CreateIndexStore.Models;
+
+#region Setup provides,indexes and seed
```
Comment (slorello89) [inline examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Program.cs:5]: remove the regions in this file :) there are limited circumstances where I'm pro-region but this file is much too small for that :)

File: examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Models/Employee.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿using Redis.OM.Modeling;
+
+namespace Redis.OM.CreateIndexStore.Models
+{
+    [Document(StorageType = StorageType.Hash)]
+    internal record Store
```
Comment (slorello89) [inline examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Models/Employee.cs:6]: move `Store` to it's own file. also change to public.

File: examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Models/Employee.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        [RedisIdField][Indexed] public int Id { get; set; }
+
+        [Indexed] public string FullAddress { get; set; } = null!;
+        [Indexed] public string Name { get; set; } = null!;
+    }
+
+    [Document(StorageType = StorageType.Json, Prefixes = new[] { "CANADA.STORE.TORONTO" }, IndexName = "Store1-idx")]
+    internal class Employee
```
Comment (slorello89) [inline examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Models/Employee.cs:15]: change to public

File: examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Models/Employee.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        [RedisIdField][Indexed] public string Id { get; set; } = null!;
+
+        [Indexed] public string FullName { get; set; } = null!;
+
+        [Indexed] public int Age { get; set; }
+
+        [Indexed] public EmploymentType EmploymentType { get; set; }
+    }
+
+    internal enum EmploymentType
```
Comment (slorello89) [inline examples/Redis.OM.CreateIndexStore/Redis.OM.CreateIndexStore/Models/Employee.cs:26]: change to public.

**Review by slorello89 — CHANGES_REQUESTED** [review]
Hi @zulander1 - thanks for opening this! Just a few minor things then I think this will be good to go.

**Comment by zulander1** [comment]
@slorello89 i've fixed everything as per your comments let me know if its okay 

thank's

**Review by slorello89 — APPROVED** [review]
LGTM 👍 thanks for the contribution! happy 🎃 hacktoberfest! 🎃

**Comment by SuzeShardlow** [comment]
Hi @zulander1, thanks for your PR!  If you are a member of our Discord server (https://discord.gg/redis), please let us know your handle so we can give you the `Hacktoberfest 2022` role.  Thanks!

**Comment by zulander1** [comment]
@SuzeShardlow there you go zulander#3305

**Comment by SuzeShardlow** [comment]
Thanks! I've assigned you the role in Discord 🙌🏽

**Pull request merged**